### DISCUSSION TITLE
issue: required prop is rejecting data that is set with setvalue

### DISCUSSION THREAD

**Discussion by abuhelos** [react-hook-form/react-hook-form#11087] [Q&A]
### Version Number

"^7.47.0"

### Codesandbox/Expo snack

https://codesandbox.io/s/holy-dust-7m87sp?file=/src/App.tsx

### Steps to reproduce

1. Add Image to the file input
2. Start adding another file but click cancel
3. onchange function should prevent file from clearning
4. Click submit

### Expected behaviour

The form will submit without required field errors

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Moshyfawn** [comment]
The native HTML file input gets cleared when you cancel the file selection dialog; this is the default behavior of the browser.

**Comment by abuhelos** [comment]
Yes I understand that, I was trying to make a workaround for it. I was saving the original file onclick and then replaced it onchange if the browser cleared the native file. I used setValue to do this.

When I console logged watch, getValues, and my submit function, all three would say that the file was there but if I made the field required it would reject submission.

In hindsight maybe it's not the best idea to try and override the browser.
